File: app/src/main/java/br/com/popularmovies/moviedetail/reviews/viewModel/factories/MovieReviewFactory.java
package br.com.popularmovies.moviedetail.reviews.viewModel.factories;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import br.com.popularmovies.moviedetail.reviews.viewModel.MovieReviewViewModel;
import br.com.popularmovies.services.movieService.source.MovieRepository;

public class MovieReviewFactory implements ViewModelProvider.Factory {
    private int movieId;
    private MovieRepository mMovieRepository;

    public MovieReviewFactory(MovieRepository mMovieRepository, int movieId) {
        this.mMovieRepository = mMovieRepository;
        this.movieId = movieId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new MovieReviewViewModel(mMovieRepository, this.movieId);
    }
}

File: app/src/main/java/br/com/popularmovies/moviedetail/reviews/viewModel/factories/MovieReviewFactory.kt
package br.com.popularmovies.moviedetail.reviews.viewModel.factories

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import br.com.popularmovies.moviedetail.reviews.viewModel.MovieReviewViewModel
import br.com.popularmovies.repositories.movie.MovieRepository
import javax.inject.Inject

class MovieReviewFactory @Inject constructor(
        private val mMovieRepository: MovieRepository,
        private val movieId: Int) : ViewModelProvider.Factory {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MovieReviewViewModel(mMovieRepository, this.movieId) as T
    }
}
